apps/web/src/hooks/useDocumentParameters/index.ts

```diff
@@ -72,28 +80,70 @@ function mapLogParametersToInputs({
 
 type InputsByDocument = Record<string, PlaygroundInputs<InputSource>>
 
+function getLinkedDataset({
+  document,
+  localInputs,
+}: {
+  document: DocumentVersion
+  localInputs: PlaygroundInputs<'dataset'>['dataset']
+}) {
+  const datasetId = document.datasetId
+  if (!datasetId) return EMPTY_LINKED_DATASET
+
+  const all = document.linkedDataset ?? {}
+  const isEmpty = Object.keys(all).length === 0
+
+  if (isEmpty) {
```

Comment:
QA with an existing local storage with mapped fields. Switch to main, set the map and back to this branch